File: math/fibonacci/dart/fibonacci.dart
void main() {
  List<int> sequence = getFibonacciSequence(20);

  for (int i = 0; i < sequence.length; i++) {
    print(sequence[i]);
  }
}

List<int> getFibonacciSequence(int n) {
  List<int> sequence = new List<int>();

  for (int i = 0; i <= n; i++) {
    sequence.add(getNextNumber(i));
  }

  return sequence;
}

int getNextNumber(int n) => n <= 2 ? 1 : getNextNumber(n - 1) + getNextNumber(n - 2);

File: math/fibonacci/Ruby/fibonacci.rb
def fibonacci (num = 0)
if num < 0
  puts "Sorry, this number is invalid!"
elsif num == 0
  return 0
elsif num == 1
  return 1
else
  return fibonacci(num-2)+fibonacci(num-1)
end

end

# Take a number from the user as input and output the nth element of fibonacci sequece
print("Tell which element of the fibonacci sequence!")
num = gets.chomp.to_i
number = fibonacci(num)
print("The " + num.to_s + "th element of fibonacci is " + number.to_s + "\n")
